### DISCUSSION TITLE
Custom Theme Tutorial

### DISCUSSION THREAD

**Discussion by aneesh1122** [fast4x/RiMusic#1831] [General]
NOTE : Sometimes you need to restart the app after changing anything for changes to take affect.
**Background 1** : This is the main UI color which is used everywhere in the app. It is also used in now-playing page as the upper part of the gradient.
![1 1](https://github.com/fast4x/RiMusic/assets/45353488/f95fff63-2da5-4de7-a9e0-0157b5725446)
.
.
.
**Background 2** : This changes the color of Navigation bar, mini-player, search bar and playlist collapsible bar in Library. It is also the background color when you choose `Background from theme`
![2 1](https://github.com/fast4x/RiMusic/assets/45353488/8e1c3ebb-c47f-437f-a9a1-19f0f84d4eaf)
![2 2](https://github.com/fast4x/RiMusic/assets/45353488/d05a9e38-a4fe-42ee-b28e-9cfde56b8445)

**Background 3** : It changes the bottom gradient of the player, queue expander, mini-player play button background, shuffle button in playlists and albums page and remaining collapsible bars in Library page.
![3 1](https://github.com/fast4x/RiMusic/assets/45353488/c5ad3b53-221a-4504-a2c0-1de8e560a4da)
![3 2](https://github.com/fast4x/RiMusic/assets/45353488/3b5cba42-e8aa-4b21-9e5f-7eec29bdd964)
![3 3](https://github.com/fast4x/RiMusic/assets/45353488/0a6d78c0-eea9-48c5-b0fc-fa3ebb4009ea)

**Background 4** : Couldn't find any changes

**Background 5** : It changes your Library icons backround(favorites, cached etc) and vacant place in playlists cover
![5 1](https://github.com/fast4x/RiMusic/assets/45353488/2f8d921d-6721-42fd-9a10-3cf0798df04b)
![5 2](https://github.com/fast4x/RiMusic/assets/45353488/d9ebaeb1-2ae0-443e-85da-e53127b66110)

**Text** : It changes text color throughout the app.

**Text secondary** : It changes color of the texts which are under other texts
![7 1](https://github.com/fast4x/RiMusic/assets/45353488/0d8cebf3-2b1a-4962-b6b8-7a05b5682ce4)

**Text disabled** : It changes color of navigation bar icons, Song and Artists' name of 'on device' songs in Player and those names which are not linked to official album/artist in YT, queue expansion bar icons and options icons not affected by ‘Text’
![8 2](https://github.com/fast4x/RiMusic/assets/45353488/c4bc2311-144b-406a-8a86-bccb2d95cf36)
![8 1](https://github.com/fast4x/RiMusic/assets/45353488/27dad046-40c3-44b2-b454-9f87deab5613)

**Icon button player** : It changes color of mini player icons.
![9 1](https://github.com/fast4x/RiMusic/assets/45353488/149be660-05e5-4846-a972-4eba7f0e2807)

**Accent** : This basically replaces ‘Green’ color of the default theme.

**Comment by fast4x** [comment]
Great work, thanks

**Comment by aneesh1122** [comment]
> Great work, thanks

You're welcome :)

what does 'Background 4' do? I couldn't find any changes

**Reply by fast4x** [reply]
i don't remember, I'll check it

There is also a Background0

**Reply by aneesh1122** [reply]
and I guess you did not add it because it changes nothing. just like Background 4

**Comment by veloal** [comment]
it would be possible to add opacity to the background color, and hex or rgb code to the custom theme options?

**Reply by fast4x** [reply]
hi, create a feature request please and add what you want

**Comment by fergassss1** [comment]
Background5 also changes this thing on quick picks
![Screenshot_2024-11-04-13-20-18-492_it fast4x rimusic-edit](https://github.com/user-attachments/assets/ce0f63c9-f957-4a1b-9664-735bfab380c0)
